co-authors-plus.php

```diff
@@ -1050,24 +1050,112 @@ function filter_wp_get_object_terms( $terms, $object_ids, $taxonomies, $args ) {
 	function filter_count_user_posts( $count, $user_id ) {
 		$user = get_userdata( $user_id );
 
-		if ( is_object( $user ) ) {
+		// Return combined post count, if account is linked.
+		if ( $this->has_linked_account( $user ) ) {
+			return $this->get_combined_post_count( $this->get_coauthor_by( 'user_nicename', $user->user_nicename ) );
+		}
+
+		// Return term count, if account is not linked and term count is available.
+		if ( $this->has_author_terms( $user ) ) {
```

Comment:
Instead of introducing an additional function call for a check to see if there is a term associated with an author, is there another way we can do so with the pre-existing code?

Reply:
I removed the additional function call and restructured the code slightly.